server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -2923,6 +2956,130 @@ public int deleteUpgradeSegmentsForTask(final String taskId)
     );
   }
 
+  @Override
+  public Set<DataSegment> findSegmentsWithUnreferencedLoadSpecs(final Set<DataSegment> segments)
+  {
+    if (segments.isEmpty()) {
+      return segments;
+    }
+    final String dataSource = segments.stream()
```

Comment:
Since we don't validate that segments don't belong to different data sources (like some methods do), we can at least call that out in the interface method

Reply:
Called out in the doc